tests/net/socket/tcp/prj.conf

```diff
@@ -57,3 +57,6 @@ CONFIG_NET_CONTEXT_SNDBUF=y
 # Because of userspace and recvmsg() we need some extra heap in order to
 # copy the iovec in the recvmsg tests.
 CONFIG_HEAP_MEM_POOL_SIZE=512
+
+# If using TF-M, disable the BL2 bootloader to save flash-space for the test.
+CONFIG_TFM_BL2=n
```

Comment:
Shouldn't this be placed to `testcase.yaml` file so that it is used only for this board?

Reply:
`boards/nrf5340dk_nrf5340_cpuapp_ns.conf` ?

Reply:
Yes, for example like that.

Reply:
It could be, but the nrf9160dk_nrf9160_ns sample, as well as all variants of these SOC are all very close, like 99% flash usage. So I thought I would just turn it off for all boards.

For boards that do not enable TF-M this is not an issue, as the configuration is off anyway.

Reply:
If you think it does not make sense to isolate the settings for multiple boards then setting it globally is certainly an option.
There is this setting in `tests/net/lib/tls_credentials/testcase.yaml`
```
   filter: CONFIG_BUILD_WITH_TFM
    extra_args: OVERLAY_CONFIG=./prj_trusted_tfm.conf
```
Would it be possible to have something like this in separate overlay file instead of enabled globally? This does not change much but at least the setting is in its own file.

Reply:
Wouldn't that mean that you would still get a compilation error when you build it yourself locally to run the test?

Reply:
Yes, but normally these are run by twister so there is no issue. Anyway, my comment was very much nit picking, if you think it does not make sense to somehow isolate the setting to certain boards then that is fine too.

Reply:
I'd prefer not to isolate this to certain boards.

Also I don't want to fix the build in the testcase.yaml.
To me it seems like that should be used to run the test in certain scenario, not to add a configuration when another configuration is enabled.
And I don't run the testcases using twister when I debug, that would confuse me a lot if I got another report like this and I couldn't reproduce it.